src/cueUtilities.js

```diff
@@ -319,60 +320,83 @@ module.exports = function (params, cy, api) {
+        function clickedOnCueRegion(node, event) {
+          const expandcollapseRenderedStartX = node._private.data.expandcollapseRenderedStartX;
+          const expandcollapseRenderedStartY = node._private.data.expandcollapseRenderedStartY;
+          const expandcollapseRenderedRectSize = node._private.data.expandcollapseRenderedCueSize;
+          const expandcollapseRenderedEndX = expandcollapseRenderedStartX + expandcollapseRenderedRectSize;
+          const expandcollapseRenderedEndY = expandcollapseRenderedStartY + expandcollapseRenderedRectSize;
+
+          return event.offsetX >= expandcollapseRenderedStartX && event.offsetX <= expandcollapseRenderedEndX &&
+                  event.offsetY >= expandcollapseRenderedStartY && event.offsetY <= expandcollapseRenderedEndY;
+        }
+
+        function getGroupOfClickedOnCue(event) {
+          for (var i = 0; i < selectedGroups.length; i++) {
+            if (clickedOnCueRegion(selectedGroups[i], event)) {
+              return selectedGroups[i];
+            }
           }
-          if (nodeHasCue){
-            var expandcollapseRenderedStartX = node._private.data.expandcollapseRenderedStartX;
-            var expandcollapseRenderedStartY = node._private.data.expandcollapseRenderedStartY;
-            var expandcollapseRenderedRectSize = node._private.data.expandcollapseRenderedCueSize;
-            var expandcollapseRenderedEndX = expandcollapseRenderedStartX + expandcollapseRenderedRectSize;
-            var expandcollapseRenderedEndY = expandcollapseRenderedStartY + expandcollapseRenderedRectSize;
-                    
-                    var cyRenderedPos = event.renderedPosition || event.cyRenderedPosition;
-            var cyRenderedPosX = cyRenderedPos.x;
-            var cyRenderedPosY = cyRenderedPos.y;
-            var factor = (opts.expandCollapseCueSensitivity - 1) / 2;
-
-            if ( (Math.abs(oldMousePos.x - currMousePos.x) < 5 && Math.abs(oldMousePos.y - currMousePos.y) < 5)
-              && cyRenderedPosX >= expandcollapseRenderedStartX - expandcollapseRenderedRectSize * factor
-              && cyRenderedPosX <= expandcollapseRenderedEndX + expandcollapseRenderedRectSize * factor
-              && cyRenderedPosY >= expandcollapseRenderedStartY - expandcollapseRenderedRectSize * factor
-              && cyRenderedPosY <= expandcollapseRenderedEndY + expandcollapseRenderedRectSize * factor) {
-              if(opts.undoable && !ur)
-                ur = cy.undoRedo({
-                  defaultActions: false
+          return null;
+        }
+
+        function cueClick(event) {
+          let node = hoveredGroup;
+          if (!node && selectedGroups.length > 0) {
+            node = getGroupOfClickedOnCue(event);
+          }
+          // no cue exists
+          if (!node) {
+            return;
+          }
+          let opts = options();
+
+          if (clickedOnCueRegion(node, event)) {
```

Comment:
Given that you call `getGroupOfClickedOnCue()` above, can this call to `clickedOnCueRegion()` ever be false?

Reply:
Yes, it is if we are clicking on an unselected group we are hovering over, but we did not click within the cue region.

Reply:
Oh, I see. Thanks for the explanation.